lib/Optimizer/Transforms/ObserveAnsatz.cpp

```diff
@@ -81,7 +81,15 @@ struct AnsatzFunctionAnalysis {
 
     // walk and find all quantum allocations
     funcOp->walk([&](quake::AllocaOp op) {
-      data.nQubits += op.getResult().getType().cast<quake::VeqType>().getSize();
+      if (auto veq = dyn_cast<quake::VeqType>(op.getResult().getType())) {
+        // Only update data.nQubits here. data.qubitValues will be updated for
+        // the corresponding ExtractRefOP's in the `walk` below.
+        data.nQubits += veq.getSize();
```

Comment:
What if `veq.hasSpecifiedSize() == false`?

Reply:
This pre-existing portion was just moved into an `if` statement. That being said - it looks like it will do `+= 0` because `getSize()` will return `0`. I can update the logic to fail the pass if that's the right thing to do. (That seems like the right thing to do to me, but let me know if you think otherwise.)

Reply:
I think we should test for that case.

Failing seems like the right thing here. Mapping when we don't know the number of qubits statically seems like it'll fall apart.

Reply:
Check for `hasSpecifiedSize()` added in ec9ad891